be/src/seekAttachment/middleware.ts

```diff
@@ -0,0 +1,79 @@
+import { Context, Middleware } from 'koa';
+import fetch from 'node-fetch';
+
+import { SEEK_API_BASE_URL } from '../constants';
+import { GetPartnerToken } from '../getPartnerToken';
+
+import { filterHeaders } from './headers';
+import { SeekAttachmentEvent, SeekAttachmentMiddlewareOptions } from './types';
+
+const wrapRetriever = async (
+  ctx: Context,
+  getPartnerToken: GetPartnerToken,
+) => {
+  const request = {
+    authorization: ctx.get('Authorization') || undefined,
+  };
+
+  try {
+    return await getPartnerToken(request);
+  } catch (err) {
+    // This is a bit of a hack. Consider either exposing the full Koa context to
+    // `getPartnerToken`, or standardising error behaviour so that we don’t
+    // handle unexpected errors and expose internal workings to the client.
+    return ctx.throw(401, err);
+  }
+};
+
+const parseUrlParameter = (ctx: Context) => {
+  const { url } = ctx.query as Record<string, unknown>;
```

Comment:
Hmm, I'm guessing this can be used to make a `GET` request to any URL on `graphql.seek.com` with a partner token? Is that existing behaviour?

Reply:
Yeah it is unfortunately. Do you have a better idea than very strict path matching on known attachment routes?

Reply:
I was thinking that partners would specify something like the `CandidateProfile` ID and attachment index and the server would request that. I can see why they wouldn't though - that would be annoying.

This seems like a good motivator for the hypothetical `download:candidate-profile-attachments` browser token scope.

Reply:
There’s more to flesh out there, like validating that the user has access to that `CandidateProfile`. I’ll go with a pattern match for now and include some comments.